sources/web/datalab/static/datalab.js

```diff
@@ -12,13 +12,17 @@
  * the License.
  */
 
+var debug = {
+  enabled: true,
+  log: function() { console.log.apply(console, arguments); }
+};
+
+function shouldShimWebsockets() {
+    return location.host.substr(-12) === '.appspot.com';
```

Comment:
The answer is probably not required, but should this be lower-cased before comparison?